### DISCUSSION TITLE
共享内存的使用问题

### DISCUSSION THREAD

**Discussion by rookiecode1126** [apache/apisix#7995] [Q&A] [answered]
在config-default.yaml中申请了多个lua_shared_dict，其中的多个共享内存是具有使用条件的。如plugin-limit-conn: 10m
![image](https://user-images.githubusercontent.com/63714578/192450287-cc5cc3ca-25bf-4733-8175-ccf8e04eab89.png)
那我的问题是：当我没有使用该插件的时候，是否会申请名为plugin-limit-conn的共享内存。

**Comment by tzssangglass** [comment, marked as answer]
Turning on this plugin will apply shared dict, you can check it in `conf/nginx.conf`

**Reply by tokers** [reply]
First of all, the plugin feature should be enabled in config.yaml, which is used to decide if the shared dict should be allocated. As for whether you use the plugin in the Route, it doesn't matter.

**Reply by rookiecode1126** [reply]
我可以这样理解么：没有在config.yaml中声明的插件，网关不会为其申请共享内存

**Reply by tzssangglass** [reply]
1. `config.yaml` configures `plugins`, as you understand it.
2. `config.yaml` is not configured with `plugins`, use the `plugins` in `config-default.yaml` as the default plugins.
